modules/renter/repair.go

```diff
@@ -154,19 +237,38 @@ func (r *Renter) threadedRepairUploads() {
 			}
 
 			// build host list
+			totalsize := f.pieceSize * uint64(f.erasureCode.NumPieces()) * f.numChunks()
 			var hosts []uploader
-			randHosts := r.hostDB.RandomHosts(f.erasureCode.NumPieces())
-			for i := range randHosts {
+			randHosts := r.hostDB.RandomHosts(f.erasureCode.NumPieces() * 2)
+			for _, h := range randHosts {
+				// probabilistically filter out known bad hosts
+				// unresponsive hosts will be selected with probability 1/(1+nFailures)
+				nFailures, ok := r.blacklist[h.IPAddress]
+				if n, _ := crypto.RandIntn(1 + nFailures); ok && n != 0 {
+					continue
+				}
+
 				// TODO: use smarter duration
-				hostUploader, err := r.newHostUploader(randHosts[i], f.size, defaultDuration, f.masterKey)
+				hostUploader, err := r.newHostUploader(h, totalsize, defaultDuration, f.masterKey)
```

Comment:
I think that `totalsize` is too big. You should only need `totalsize` / `numPieces`